source/opt/ir_context.cpp

```diff
@@ -1043,5 +1043,25 @@ bool IRContext::IsReachable(const opt::BasicBlock& bb) {
   return GetDominatorAnalysis(enclosing_function)
       ->Dominates(enclosing_function->entry().get(), &bb);
 }
+
+SpvExecutionModel IRContext::GetExecutionModel() {
```

Comment:
I would remove this from IRContext and put it back in the only pass it was used in previously.